homeassistant/components/shelly/utils.py

```diff
@@ -497,3 +501,53 @@ def async_remove_shelly_rpc_entities(
 def is_rpc_thermostat_mode(ident: int, status: dict[str, Any]) -> bool:
     """Return True if 'thermostat:<IDent>' is present in the status."""
     return f"thermostat:{ident}" in status
+
+
+def get_virtual_component_ids(config: dict[str, Any], platform: str) -> list[str]:
+    """Return a list of virtual component IDs for a platform."""
+    ids: list[str] = []
+    if platform == Platform.SWITCH.value:
+        ids.extend(
+            k
+            for k, v in config.items()
+            if k.startswith("boolean") and v["meta"]["ui"]["view"] == "toggle"
+        )
+    if platform == Platform.BINARY_SENSOR.value:
+        ids.extend(
+            k
+            for k, v in config.items()
+            if k.startswith("boolean") and v["meta"]["ui"]["view"] == "label"
+        )
+
+    return ids
```

Comment:
Since the method here duplicates the same logic and the only change is `toggle` or `label` I think it would be better to remove the condition on `Platform` parameter and pass `toggle` when we call this parameter from the `SWITCH` platform and `label` when we call it from the `BINARY_SENSOR` platform

Another option to is to split into two methods - `get_switch_virtual_component_ids` and `get_binary_sensor_virtual_component_ids`